lib/engine.js

```diff
@@ -292,7 +293,11 @@ module.exports = function container (get, set, clear) {
               }
               if (order.status === 'rejected' && (order.reject_reason === 'post only' || api_order.reject_reason === 'post only')) {
                 msg('post-only ' + type + ' failed, re-ordering')
-                return cancelOrder(true)
+                return cb(null, null)
+              }
+              if (order.status === 'rejected' && order.reject_reason === 'balance') {
```

Comment:
As we removed previous line 285, we need to correct this to `if (order.status === 'rejected' & (order.reject_reason === 'balance' || api_order.reject_reason === 'balance')) {`